builtin/logical/database/version_wrapper.go

```diff
@@ -24,6 +25,12 @@ var _ logical.PluginVersioner = databaseVersionWrapper{}
 // newDatabaseWrapper figures out which version of the database the pluginName is referring to and returns a wrapper object
 // that can be used to make operations on the underlying database plugin.
 func newDatabaseWrapper(ctx context.Context, pluginName string, pluginVersion string, sys pluginutil.LookRunnerUtil, logger log.Logger) (dbw databaseVersionWrapper, err error) {
+	// 1.12.0 and 1.12.1 stored plugin version in the config, but that stored
+	// builtin version may disappear from the plugin catalog when Vault is
+	// upgraded, so always reference builtin plugins by an empty version.
+	if versions.IsBuiltinVersion(pluginVersion) {
```

Comment:
Wonder if it is worth updating the docstring to mention that a built-in`pluginVersion` is "ignored"